lms/djangoapps/certificates/tests/test_badge_handler.py

```diff
@@ -0,0 +1,173 @@
+"""
+Tests for the BadgeHandler, which communicates with the Badgr Server.
+"""
+from django.test.utils import override_settings
+from eventtracking import tracker
+from lazy.lazy import lazy
+from mock import patch, Mock, call
+from certificates.models import BadgeAssertion
+from xmodule.modulestore.tests.factories import CourseFactory
+from certificates.badge_handler import BadgeHandler
+from student.tests.factories import UserFactory, CourseEnrollmentFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, TEST_DATA_DIR
+
+BADGR_SETTINGS = {
+    'BADGR_API_TOKEN': '12345',
+    'BADGR_BASE_URL': 'https://example.com',
+    'BADGR_ISSUER_SLUG': 'test-issuer',
+    'BADGR_IMAGE_SOURCES': {
+        'honor': TEST_DATA_DIR / "badges" / "bulb.svg"
+    }
+}
+
+
+class BadgeHandlerTestCase(ModuleStoreTestCase):
+    """
+    Tests the BadgeHandler object
+    """
+    def setUp(self):
+        """
+        Create a course and user to test with.
+        """
+        super(BadgeHandlerTestCase, self).setUp()
+        # Need key to be deterministic to test slugs.
+        self.course = CourseFactory.create(org='edX', course='course_test', run='test_run', display_name='Badged')
+        self.user = UserFactory.create(email='example@example.com')
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.location.course_key, mode='honor')
+        # Need for force empty this dict on each run.
+        BadgeHandler.badges = {}
+
+    @lazy
+    def handler(self):
+        """
+        Lazily loads a BadgeHandler object for the current course. Can't do this on setUp because the settings
+        overrides aren't in place.
+        """
+        return BadgeHandler(self.course.location.course_key)
+
+    @override_settings(**BADGR_SETTINGS)
```

Comment:
Optional: I believe that it is safe to override_settings for the whole class and they should apply for each test